### PR TITLE
WIP: Updating DataSource implementations for spark 2.4.

### PR DESCRIPTION
**Pull request by mmn599**
Reference change.

### CODE REVIEW

File: src/main/scala/com/microsoft/cdm/write/CDMDataWriterFactory.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ class CDMDataWriterFactory(var adlProvider: ADLGen2Provider,
 
   // TODO: error handling. we're basically assuming successful writes. Need to add logic to remove/rewrite files on failure.
 
-  def createDataWriter(partitionId: Int, attemptNumber: Int): DataWriter[Row] = {
+  def createDataWriter(partitionId: Int, taskId: Long, epochId: Long): DataWriter[Row] = {
```
Comment (RobertBruckner) [inline src/main/scala/com/microsoft/cdm/write/CDMDataWriterFactory.scala]: Looks like these arguments are not used inside?

**Comment by adiazcan** [comment]
When this version will be available? I can't create a Spark 2.3 cluster on Azure Databricks

File: src/main/scala/com/microsoft/cdm/write/CDMDataWriterFactory.scala (hunk 1 above)
Comment (adiazcan) [reply src/main/scala/com/microsoft/cdm/write/CDMDataWriterFactory.scala]: Any change to have this new version next week?

**Comment by YuHe18** [comment]
Hi, are there any updates on this issue?

**Pull request merged**